kanin/src/app.rs

```diff
@@ -112,7 +120,28 @@ impl App {
     /// # Panics
     /// On connection errors after the initial connection is established, the app will simply panic.
     pub async fn run(self, amqp_addr: &str) -> Result<()> {
-        let handles = self.setup_handlers(amqp_addr).await?;
+        debug!("Connecting to AMQP on address: {amqp_addr:?} ...");
+        let conn = Connection::connect(amqp_addr, ConnectionProperties::default()).await?;
+        trace!("Connected to AMQP on address: {amqp_addr:?}");
+        self.run_with_connection(&conn).await
+    }
+
+    /// Runs the app with all the handlers that have been registered.
+    ///
+    /// Each handler is given its own dedicated channel associated with the single connection created from the provided connection.
+    /// The handlers then run in their own spawned tokio tasks.
+    /// Handlers handle requests concurrently by spawning new tokio tasks for each incoming request.
+    ///
+    /// # Errors
+    /// Returns an `Err` on any of the below conditions:
+    /// * No handlers were registered.
+    /// * A connection to the AMQP broker could not be established.
+    /// * Queue/consumer declaration or binding failed while setting up a handler.
+    ///
+    /// # Panics
+    /// On connection errors after the initial connection is established, the app will simply panic.
```

Comment:
This function no longer performs this initial connecting so this comment doesn't make as much sense.